kairon/actions/definitions/form_validation.py

```diff
@@ -70,6 +73,10 @@ async def execute(self, dispatcher: CollectingDispatcher, tracker: Tracker, doma
 
             if is_valid:
                 status = "SUCCESS"
+                if custom_value and form_slot_set_type == FORM_SLOT_SET_TYPE.CUSTOM.value:
+                    slot_value = custom_value
+                elif slot and form_slot_set_type == FORM_SLOT_SET_TYPE.SLOT.value:
+                    slot_value = slot
```

Comment:
slot_value = tracker.get_slot(validation.slot_set.value)
You want the slot value to be taken from another slot if form_slot_set_type is slot.